featuretools/utils/recommend_primitives.py

```diff
@@ -0,0 +1,264 @@
+    "is_whole_number",
+    "not_equal",
+    "multiply_numeric",
+    "multiply_numeric_boolean",
+    "add_numeric",
+    "less_than_equal_to",
+    "divide_numeric",
+    "less_than",
+    "subtract_numeric",
+    "modulo_numeric",
+    "cosine",
+    "tangent",
+    "sine",
+]
+
+DEFAULT_EXCLUDED_PRIMITIVES = (
+    REQUIRED_INPUT_PRIMITIVES
+    + DEPRECATED_PRIMITIVES
+    + ORDERED_PRIMITIVES
+    + OTHER_PRIMITIVES_TO_EXCLUDE
+)
+
+TIME_SERIES_PRIMITIVES = [
+    "rolling_count",
+    "rolling_max",
+    "rolling_mean",
+    "rolling_min",
+    "rolling_std",
+    "rolling_trend",
+    "lag",
+]
+
+
+# TODO: Support multi-table
+def get_recommended_primitives(
+    entityset: EntitySet,
+    target_dataframe_name: str,
+    is_time_series: bool,
+    excluded_primitives: List[str] = DEFAULT_EXCLUDED_PRIMITIVES,
+) -> List[str]:
+    """Get a list of recommended primitives given an entity set.
+
+    Description:
+        Given a single table entity set with a `target_dataframe_name`, `is_time_series` specified
+        and a list of any primitives in `excluded_primitives` to not be included in the final recommendation list.
+
+    Args:
+        entityset (EntitySet): EntitySet that only contains one dataframe.
+        target_dataframe_name (str): Name of target dataframe to access in `entityset`.
+        is_time_series (bool): Whether or not time-series analysis will be performed. If set to `True`, `Lag` will always be recommended,
+        as well as all Rolling primitives if numeric columns are present.
+        excluded_primitives (List[str]): List of transform primitives to exclude from recommendations.
+
+    Note:
+        This function currently only works for single table and will only recommend transform primitives.
+    """
+    recommended_primitives = set()
+
+    if not is_time_series:
+        excluded_primitives += TIME_SERIES_PRIMITIVES
+
+    if is_time_series:
+        recommended_primitives.add("lag")
+
+    all_trans_primitives = get_transform_primitives()
+    selected_trans_primitives = [
+        p for name, p in all_trans_primitives.items() if name not in excluded_primitives
+    ]
+
+    valid_primitive_names = [
+        prim.name
+        for prim in get_valid_primitives(
+            entityset,
+            target_dataframe_name,
+            1,
+            selected_trans_primitives,
+        )[1]
+    ]
+
+    recommended_primitives.update(
+        _recommend_non_numeric_primitives(
+            entityset,
+            target_dataframe_name,
+            valid_primitive_names,
+        ),
+    )
+    skew_numeric_primitives = set(["square_root", "natural_logarithm"])
+    valid_skew_primtives = skew_numeric_primitives.intersection(valid_primitive_names)
+
+    time_series_primitives = set(TIME_SERIES_PRIMITIVES)
+    valid_time_series_primitives = time_series_primitives.intersection(
+        valid_primitive_names,
+    )
+
+    numeric_tags_only_df = entityset[target_dataframe_name].ww.select("numeric")
```

Comment:
Would it be better to move this selection process into the `_recommend_skew_numeric_primitives` function?

Reply:
I didn't want to add the extra parameters to `_recommend_skew_numeric_primitives` if we didn't need them, also we may end up using the numeric for something else in the future.